packages/core/src/bodl/bodl-emitter-service.ts

```diff
@@ -148,5 +207,25 @@ export default class BodlEmitterService implements BodlService {
         ];
     }
 
+    private trackCompletedStep(step: AnalyticStepType) {
+        this._completedSteps[step] = true;
+        this.bodlEvents.emit('bodl_checkout_step_completed', { step });
+    }
+
+    private hasStepCompleted(step: AnalyticStepType): boolean {
+        return this._completedSteps[step];
+    }
+
+    private backfillCompletedSteps(currentStep: AnalyticStepType): void {
+        for (const step of AnalyticStepOrder) {
```

Comment:
Do we need to use `for...of` here?

Reply:
After discussion - will change to `forEach`